src/session/sessionReducer.js

```diff
@@ -19,6 +19,41 @@ import {
 } from '../actionConstants';
 import { getHasAuth } from '../account/accountsSelectors';
 
+/**
+ * Miscellaneous non-persistent state specific to a particular account.
+ *
+ * See {@link SessionState} for discussion of what "non-persistent" means.
+ */
+export type PerAccountSessionState = $ReadOnly<{
+  eventQueueId: number,
+
+  /**
+   * Whether the /register request is in progress.
+   *
+   * This happens on startup, or on re-init following a dead event
+   * queue after 10 minutes of inactivity.
+   */
+  loading: boolean,
+
+  needsInitialFetch: boolean,
+
+  outboxSending: boolean,
+
+  /**
+   * Whether `ServerCompatNotice` (which we'll add soon) has been
+   *   dismissed this session.
+   *
+   * We put this in the per-session state deliberately, so that users
+   * see the notice on every startup until the server is upgraded.
+   * That's a better experience than not being able to load the realm
+   * on mobile at all, which is what will happen soon if the user
+   * doesn't act on the notice.
+   */
+  hasDismissedServerCompatNotice: boolean,
```

Comment:
Cool! This looks like progress toward solving a pretty small bug, but a bug nonetheless.

From https://github.com/zulip/zulip-mobile/pull/4750#issue-892286064:

> NOTE: The has-dismissed-the-banner state isn't stored per-realm, so there's currently a bug where dismissing the banner makes it go away for all <2.0.0 realms the device knows about (until app restart, anyway). I figured this was minor enough to be acceptable for the time being, unless we can think of a nice and easy way to store the state both per-realm and per-session, while being sure to clean up after ourselves when the user wants to forget the realm.

Reply:
(I think per-account is a totally fine proxy for per-realm.)